### ISSUE TITLE
Way to bypass message about uncommitted and gitignored files?

### ISSUE THREAD

**Issue by dmwyatt** [caprover/caprover-cli#62]
Is there a way to bypass message about uncommitted and gitignored files?  I don't see anything relevant in `captover deploy --help`.  By "bypass" I'd just like to be able to automatically answer yes or no in the same way I can provide the (for example) app name via the `-a` command line switch.

**Comment by githubsaturn** [comment]
The deploy pipeline is fully customizable. By default CapRover uses `git archive` which ignores uncommitted and ignored files. But you can build your own tar file, for example:

```
tar -cvf ./deploy.tar  ./build/*
caprover deploy -t ./deploy.tar
```
See a real example and use case here: https://caprover.com/docs/recipe-deploy-create-react-app.html

**Comment by dmwyatt** [comment]
Yes, this is how I do it currently.

It still seems like there should be a flag to bypass this message. I
believe all other prompts can be bypassed with command line flags, no?

**Comment by dmwyatt** [comment]
Right. What I'm trying to get at is this:

Is that a useful distinction?  

Does `caprover-cli` not allow bypassing that prompt because a project goal says "do not bypass prompts for external tools" or something that has that effect or does it not allow bypassing the prompt because no one has got around to coding it?

**Comment by githubsaturn** [comment]
It's by intention. CapRover design follows thin interface layer for more complicated tools. 

For example, in this case, there are tens of different flags that are available with `tar` CLI. We certainly don't want to pollute CapRover API with all these flags. CapRover takes care of the most common case, if further customization is required, users can choose to do so.

For example, you're asking to include uncommitted files, another user might be asking to only include a specific directory. A third user asks for two directories. The next user asks for two directories, but exclude `*.map` files. Porting all these use cases to CapRover is just a costly, yet not useful, approach and increases maintenance cost of CapRover drastically. However, instead of porting all these use cases, if the most common use case does not satisfy someone's needs, CapRover allows them to customize their pipeline. In this case, it translates to manually creating tarball and ask CapRover to push the tarball to the server.

**Comment by dmwyatt** [comment]
Yeah, that's certainly a reasonable project philosophy and something I've done on many projects over the last couple of decades.

However, I just want to point out that (i think! I'm not super familiar with the codebase) in this instance, caprover-cli is specifically [doing a check](https://github.com/caprover/caprover-cli/blob/5e2ea42a84095a4d6f82541ab84531b64d8ea60e/src/commands/deploy.ts#L185) and prompting for uncommitted and ignored files.  If I just run the [underlying ](https://github.com/caprover/caprover-cli/blob/1a3d74b4856f384cd85258704fd6fdfdf785244f/src/utils/DeployHelper.ts#L102)`git archive` command manually, there is no prompting.

In other words, in this instance, `caprover-cli` is actually *adding* to the underlying tool UI.

I think probably in this instance the prompt is the wrong "default" case.  That's how ignored and uncommitted files **work** in many deployment tools and git in general and the prompt feels redundant whilst adding more code to the project.
